server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyHttpRequest.java

```diff
@@ -0,0 +1,417 @@
+    @Override
+    public ResteasyAsynchronousContext getAsyncContext() {
+        return executionContext;
+    }
+
+    @Override
+    public void forward(String path) {
+
+        // TODO
+    }
+
+    @Override
+    public boolean wasForwarded() {
+        // TODO
+        return false;
+    }
+
+    @Override
+    public String getRemoteAddress() {
+        // TODO
+        return null;
+    }
+
+    @Override
+    public String getRemoteHost() {
+        // TODO
+        return null;
+    }
+
+    class NettyExecutionContext extends AbstractExecutionContext {
+        protected final ReactorNettyHttpRequest request;
+        protected final ReactorNettyHttpResponse response;
+        protected volatile boolean done;
+        protected volatile boolean cancelled;
+        protected volatile boolean wasSuspended;
+        protected NettyExecutionContext.NettyHttpAsyncResponse asyncResponse;
+
+        NettyExecutionContext(final ReactorNettyHttpRequest request, final ReactorNettyHttpResponse response, final SynchronousDispatcher dispatcher)
+        {
+            super(dispatcher, request, response);
+            this.request = request;
+            this.response = response;
+            this.asyncResponse = new NettyExecutionContext.NettyHttpAsyncResponse(dispatcher, request, response);
+        }
+
+        @Override
+        public boolean isSuspended() {
+            return wasSuspended;
+        }
+
+        @Override
+        public ResteasyAsynchronousResponse getAsyncResponse() {
+            return asyncResponse;
+        }
+
+        @Override
+        public ResteasyAsynchronousResponse suspend() throws IllegalStateException {
+            return suspend(-1);
+        }
+
+        @Override
+        public ResteasyAsynchronousResponse suspend(long millis) throws IllegalStateException {
+            return suspend(millis, TimeUnit.MILLISECONDS);
+        }
+
+        @Override
+        public ResteasyAsynchronousResponse suspend(long time, TimeUnit unit) throws IllegalStateException {
+            if (wasSuspended)
+            {
+                throw new IllegalStateException("oh nos!");
+            }
+            wasSuspended = true;
+            return asyncResponse;
+        }
+
+        @Override
+        public void complete() {
+            if (wasSuspended) {
+                asyncResponse.complete();
+            }
+        }
+
+        @Override
+        public CompletionStage<Void> executeAsyncIo(CompletionStage<Void> f) {
+            // check if this CF is already resolved
+            CompletableFuture<Void> ret = f.toCompletableFuture();
+            // if it's not resolved, we may need to suspend
+            if(!ret.isDone() && !isSuspended()) {
+                suspend();
+            }
+            return ret;
+        }
+
+        @Override
+        public CompletionStage<Void> executeBlockingIo(RunnableWithException f, boolean hasInterceptors) {
+            if(1 == 1) { // TODO if(!NettyUtil.isIoThread()) {
```

Comment:
As with a lot of this code, it is just copied from netty and needs to be reviewed.  Before I rolled off of this, I was trying to understand when this method would be called.  I found the conditional in resteasy code, but I wasn't sure how to get into the state such that this would be called.